crates/curp/src/server/raw_curp/log.rs

```diff
@@ -82,21 +82,28 @@ impl<C: Command> FallbackContext<C> {
 struct LogEntryVecDeque<C: Command> {
     /// A VecDeque to store log entries, it will be serialized and persisted
     entries: VecDeque<Arc<LogEntry<C>>>,
-    /// The sum of serialized size of previous log entries
-    /// batch_index[i+1] = batch_index[i] + size(entries[i])
-    batch_index: VecDeque<u64>,
+    /// entry size of each item in entries
+    entry_size: VecDeque<u64>,
+    /// the right index of the batch (offset)
+    /// batch_range: [i, i + batch_index[i]]
```

Comment:
Yes, I record the relative index of each element, which simplifies the process of compact because the relevant index is the same when calling `pop_front`.

Reply:
Why don't you use the `li_to_pi` method to convert a logic index to a physical index?

Reply:
I leave an issue #800  to track